code/datums/components/pricetag.dm

```diff
@@ -0,0 +1,21 @@
+/datum/component/pricetag
+	var/datum/bank_account/owner = null
+	var/profit_ratio = 1
+
+/datum/component/pricetag/Initialize(_owner,_profit_ratio)
+	if(_owner)
+		owner = _owner
```

Comment:
You should be checking if the parent is an item and returning `COMPONENT_INCOMPATIBLE` if not